backend/api/v1/rollout_service.go

```diff
@@ -73,6 +73,63 @@ func (s *RolloutService) GetPlan(ctx context.Context, request *v1pb.GetPlanReque
 	return convertToPlan(plan), nil
 }
 
+// ListPlans lists plans.
+func (s *RolloutService) ListPlans(ctx context.Context, request *v1pb.ListPlansRequest) (*v1pb.ListPlansResponse, error) {
+	projectID, err := getProjectID(request.Parent)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	var limit, offset int
+	if request.PageToken != "" {
+		var pageToken storepb.PageToken
+		if err := unmarshalPageToken(request.PageToken, &pageToken); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "invalid page token: %v", err)
+		}
+		if pageToken.Limit != request.PageSize {
+			return nil, status.Errorf(codes.InvalidArgument, "request page size does not match the page token")
+		}
+		limit = int(pageToken.Limit)
+		offset = int(pageToken.Offset)
+	}
+	if limit <= 0 {
+		limit = 50
+	}
+	if limit > 1000 {
+		limit = 1000
+	}
+	limitPlusOne := limit + 1
```

Comment:
Any reason we need to do plus one?

Reply:
Check if there's more

Reply:
Add a comment about how limitPlusOne plays the trick. LG otherwise.